lib/library.rb

```diff
@@ -0,0 +1,68 @@
+class Library
+  attr_reader :name,
+              :books,
+              :authors,
+              :checked_out_books,
+              :times_checked_out
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @times_checked_out = {}
+  end
+
+  def add_author(author)
+    @authors << author
+    books << author.books
+    books.flatten!
+  end
+
+  def publication_time_frame_for(author)
+    book_dates = book_dates_for(author)
+    time_frame(book_dates)
+  end
+
+  def time_frame(book_dates)
+    time_frame = {}
+    time_frame[:start] = book_dates.min
+    time_frame[:end] = book_dates.max
+    time_frame
+  end
+
+  def book_dates_for(author)
+    books = @books.select do |book|
```

Comment:
Couldn't you just do author.books and then you'd have the books that this author wrote?